cmd/virt-launcher/virt-launcher.go

```diff
@@ -610,16 +615,31 @@ func RemoveContents(dir string) error {
 
 func terminateIstioProxy() {
 	if istioProxyPresent() {
-		resp, err := http.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
-		if err != nil || resp.StatusCode != http.StatusOK {
-			log.Log.Error("Failed to request Istio proxy termination")
+		backoff := wait.Backoff{
+			Steps:    3,
+			Duration: 100 * time.Millisecond,
+			Factor:   1.0,
+			Jitter:   0.1,
+		}
+		isError := func(err error) bool {
```

Comment:
I'd rather have the concept of a retriable error - e.g. connection reset / service unavailable / (possibly) connection refused. 

You can use https://github.com/kubernetes/apimachinery/blob/e1b4d3faae10bda8d956188103745b1fecd462fe/pkg/util/net/util.go#L41 / https://github.com/kubernetes/apimachinery/blob/e1b4d3faae10bda8d956188103745b1fecd462fe/pkg/api/errors/errors.go#L536